### PR TITLE
Explorer in out registrations

### LINKED ISSUE #1686: Explorer display in/out SPOs in registered signers page
## Why
We need a new page in the explorer which could could display the list of in/out SPOs for the epoch where data is available on the aggregator (`4` epochs).

## What
Implement a new page for in/out SPOs in the explorer.

## How

- [x] Add a new in/out SPOs page in explorer
- [x] Add a button on the page to copy the out list in a Discord friendly markdown

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR add a new page to the explorer: "In/Out Registrations".

This page show for the last four epochs the signers that have joined us or are returning and the one that left us. It also have a button that show a popup with a discord friendly formatted list of movements (useful for announcement of missing registrations).

### Demo

[Screencast from 23-05-2024 18:27:45.webm](https://github.com/input-output-hk/mithril/assets/790521/db73c8f7-32b2-478a-92e4-d6d8b7dc5f6e)

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Issue(s)
Closes #1686

### CODE REVIEW

File: mithril-explorer/src/app/registrations-in-out/RegistrationsMovementsList.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const filteredRegistrations = Object.entries(registrations)
+    .reverse()
+    .filter(([_, movements]) => movementsForCurrentMode(movements).length > 0);
+
+  return (
+    <>
+      <h3>
+        {mode === "out" ? (
+          <>
+            <i className="bi bi-box-arrow-left"></i> Missing
```
Comment (jpraynaud) [inline mithril-explorer/src/app/registrations-in-out/RegistrationsMovementsList.js:26]: Could we name missing signers with `De-registered signers` and new signers with `Newly registered signers`?

File: mithril-explorer/src/components/RegistrationDiscordFormatModal.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  return (
+    <Modal
+      show={mode !== undefined}
+      onHide={handleModalClose}
+      size="xl"
+      aria-labelledby="contained-modal-title-vcenter"
+      centered>
+      <Modal.Header closeButton>
+        <Modal.Title>
+          <i className="bi bi-discord"></i> Discord formatted table of{" "}
```
Comment (jpraynaud) [inline mithril-explorer/src/components/RegistrationDiscordFormatModal.js:71]: Could we make it a bit more generic?
```suggestion
          <i className="bi bi-discord"></i> Markdown formatted table of{" "}
```

**Review by jpraynaud — APPROVED** [review]
Nice work! LGTM :+1:

File: mithril-explorer/src/components/RegistrationDiscordFormatModal.js (hunk 2 above)
Comment (Alenar) [reply mithril-explorer/src/components/RegistrationDiscordFormatModal.js:71]: Oh looks like Discord updated their markdown support some time ago and [now support more tags](https://support.discord.com/hc/en-us/articles/210298617-Markdown-Text-101-Chat-Formatting-Bold-Italic-Underline).
Before it discord only support bold, italic, and crossed text, all other markdown features were ignored. That's why I though that specifying it was discord compatible mattered.

I will do that and add  on the pool ticker the CExplorer link if available,.

Comment (Alenar) [reply mithril-explorer/src/components/RegistrationDiscordFormatModal.js:71]: Well adding the CExplorer link to the pool ticker in the markdown make it really easy to exceed discord message size limit so I won't be able to do that sadly (or we would need to split the message in several parts ... or pay discord nitro).

Comment (jpraynaud) [reply mithril-explorer/src/components/RegistrationDiscordFormatModal.js:71]: It's fine without the link :+1: 
I forgot that we could reach the message size limit with Discord :man_shrugging:

**Review by dlachaume — APPROVED** [review]
LGTM 🚀 

Just a UX suggestion:
It would be nice to have a confirmation that the text has been successfully copied when clicking on the 'Copy to clipboard' button.

**Pull request merged**